tests/chainer_tests/function_hooks_tests/test_debug_print.py

```diff
@@ -1,72 +1,138 @@
+import re
 import unittest
 
+import six
+
 import chainer
 from chainer import cuda
 from chainer import function_hooks
-from chainer import functions
-from chainer import gradient_check
-from chainer import links
 from chainer import testing
 from chainer.testing import attr
 import numpy
 
 
-class TestPrintHookToLink(unittest.TestCase):
-
-    def setUp(self):
-        self.h = function_hooks.PrintHook()
-        self.l = links.Linear(5, 5)
-        self.x = numpy.random.uniform(-0.1, 0.1, (3, 5)).astype(numpy.float32)
-        self.gy = numpy.random.uniform(-0.1, 0.1, (3, 5)).astype(numpy.float32)
-
-    def test_name(self):
-        self.assertEqual(self.h.name, 'PrintHook')
+class DummyFunction(chainer.Function):
 
-    def test_forward_cpu(self):
-        with self.h:
-            self.l(chainer.Variable(self.x))
+    def forward(self, inputs):
+        self.retain_inputs((0,))
+        return inputs[0],
 
-    @attr.gpu
-    def test_forward_gpu(self):
-        self.l.to_gpu()
-        with self.h:
-            self.l(chainer.Variable(cuda.to_gpu(self.x)))
-
-    def test_backward_cpu(self):
-        with self.h:
-            gradient_check.check_backward(self.l, self.x, self.gy)
-
-    @attr.gpu
-    def test_backward_gpu(self):
-        self.l.to_gpu()
-        with self.h:
-            gradient_check.check_backward(
-                self.l, cuda.to_gpu(self.x), cuda.to_gpu(self.gy))
+    def backward(self, inputs, grads):
+        return (grads[0],) + (None,) * (len(inputs) - 1)
 
 
 class TestPrintHookToFunction(unittest.TestCase):
 
     def setUp(self):
-        self.h = function_hooks.PrintHook()
-        self.f = functions.Exp()
-        self.f.add_hook(self.h)
+        self.io = six.StringIO()
+        self.h = function_hooks.PrintHook(file=self.io)
+        self.f = DummyFunction()
         self.x = numpy.random.uniform(-0.1, 0.1, (3, 5)).astype(numpy.float32)
         self.gy = numpy.random.uniform(-0.1, 0.1, (3, 5)).astype(numpy.float32)
 
     def test_forward_cpu(self):
-        self.f(chainer.Variable(self.x))
+        self.f.add_hook(self.h)
+        self.f(chainer.Variable(self.x), chainer.Variable(self.x))
+        expect = '''^function\tDummyFunction
+input data
+<variable at 0x[0-9a-f]+>
+- device: CPU
+- backend: <(type|class) 'numpy.ndarray'>
+- shape: \(3, 5\)
```

Comment:
The tests on Win+Py2.7 do not pass, maybe due to this line (it is printed as `(3L, 5L)` in this environment). Can you fix it? It's enough to just fix these test cases (including this line and others below).